RichString.c

```diff
@@ -20,18 +20,25 @@ in the source distribution for its full text.
 #define charBytes(n) (sizeof(CharType) * (n))
 
 static void RichString_extendLen(RichString* this, int len) {
-   if (this->chlen <= RICHSTRING_MAXLEN) {
+   if (this->chptr == this->chstr) {
+      // String is in internal buffer
       if (len > RICHSTRING_MAXLEN) {
+         // Copy from internal buffer to allocated string
          this->chptr = xMalloc(charBytes(len + 1));
          memcpy(this->chptr, this->chstr, charBytes(this->chlen));
```

Comment:
Does not ensure trailing \NUL does it?

Reply:
`RichString_setChar(this, len, 0);` on line 38 will do this

Reply:
Exactly. NUL termination is handled below the conditional re-allocations in a central place.